lms/djangoapps/grades/signals/handlers.py

```diff
@@ -124,21 +113,32 @@ def recalculate_subsection_grade_handler(sender, **kwargs):  # pylint: disable=u
        - usage_id: Unicode string indicating the courseware instance
     """
     try:
-        course_id = kwargs.get('course_id', None)
-        usage_id = kwargs.get('usage_id', None)
-        student = kwargs.get('user', None)
+        course_id = kwargs['course_id']
+        usage_id = kwargs['usage_id']
+        student = kwargs['user']
+    except KeyError:
+        log.exception(
```

Comment:
Good point.  Right now the error handling of all the SCORE_CHANGED signal handlers are inconsistent.  One catches and logs errors, the other lets errors propagate up, and yet another (this case) catches some exceptions but propagates others.

The contract with django signals is that the sender of the signal can decide how to handle errors by calling either `send` or `send_robust`.  The former stops the signal as soon as one handler raises an exception.  The other allows all handlers to be called regardless if one or more handlers raised an error.  

In this case, we don't expect missing fields in the kwargs as the signal sender and handler are all in one process - and this layer of defense can mask other underlying issues.  So unless someone comes up with a good reason, I would vote for removing the unneeded error handling and letting unexpected errors propagate back up to the sender of the signal.